clients/drcachesim/scheduler/scheduler.h

```diff
@@ -569,19 +569,28 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
         /**
          * Returns the ordinal for the current input stream feeding this output stream.
          */
-        input_ordinal_t
+        virtual input_ordinal_t
         get_input_stream_ordinal()
         {
             return scheduler_->get_input_ordinal(ordinal_);
         }
         /**
-         * Returns the value of the last seen #TRACE_MARKER_TYPE_TIMESTAMP marker.
+         * Returns the value of the most recently seen #TRACE_MARKER_TYPE_TIMESTAMP
+         * marker.
          */
         uint64_t
         get_last_timestamp() const override
         {
             return last_timestamp_;
         }
+        /**
+         * Returns the value of the first seen #TRACE_MARKER_TYPE_TIMESTAMP marker.
```

Comment:
See above on the first for this output not being what the analyzers want.

Reply:
Fixed.